File: src/converter/inheritanceModifier.ts
import {Node} from "typescript"
import {ConverterContext} from "./context.js";
import {Signature} from "./plugins/convertParameterDeclaration.js";

export interface InheritanceModifierContext extends ConverterContext {
    signature?: Signature | undefined
    getter?: boolean | undefined
    setter?: boolean | undefined
}

export type InheritanceModifier<TNode extends Node = Node> =
    (node: TNode, context: InheritanceModifierContext) => string | null

File: gradle-plugin/karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/extension/InheritanceModifier.kt
package io.github.sgrishchenko.karakum.extension

import io.github.sgrishchenko.karakum.extension.plugins.Signature
import seskar.js.JsNative
import typescript.Node

external interface InheritanceModifierContext : ConverterContext {
    val signature: Signature?
    val getter: Boolean?
    val setter: Boolean?
}

external interface InheritanceModifier<in TNode : Node> {
    @JsNative
    operator fun invoke(node: TNode, context: InheritanceModifierContext): String?
}
